controller/controller.go

```diff
@@ -76,6 +90,9 @@ func (c *EgressController) Run(ctx context.Context) {
 				log.Errorf("Failed to ensure configuration: %v", err)
 				continue
 			}
+
+			// successfully synced
+			lastSyncTimestamp.SetToCurrentTime()
```

Comment:
This should also happen in the case above: `<-time.After(c.interval)` because this is where we resync will happen at a regular interval, otherwise we can risk the synctimestamp metric is very old if no configmaps are changed/updated/deleted.